[#109] 휴지통 화면 구현

gustn3965 opened the pull request:
## 개요

휴지통 화면 구현 

## 시연 
![삭제](https://user-images.githubusercontent.com/48749182/137321553-57b91e68-b8c0-448a-8f19-3c85c90d4289.gif)

## 작업 사항
크게, 휴지통화면에 필요한 뷰들 순서대로 구현하여 커밋을 진행했습니다.
1. 휴지통화면 셀 구현
    - `ContentsCollectionViewCell`를 리팩토링하여 휴지통화면에 필요한 뷰들을 추가했다
    -  ![image](https://user-images.githubusercontent.com/48749182/137322054-e50fe021-f6db-472c-82cd-edf597e57bed.png)


2. 휴지통 화면의 상단 헤더뷰 구현 ( `TwoLabelVerticalHeaderView`)
    - **개의 게시물, 게시물삭제 관련 내용을 포함하는 뷰다. 
    -  ![image](https://user-images.githubusercontent.com/48749182/137322035-1ca2a281-6143-4dca-ad7a-e0a7452df210.png)


3. 휴지통 화면의 하단에 나타나는 삭제 or 복구 버튼 구현
    - `LeftRightButtonView` - 좌측, 우측에 각각 버튼을 가지는 뷰다
    -  ![image](https://user-images.githubusercontent.com/48749182/137322076-00168913-afe3-4368-bd03-b6379f380058.png)


4. 휴지통 화면 구현 ( ViewController에 배치 ) 
    - `TrashViewController` 를 만들어, 휴지퉁화면에 필요한 뷰들을 배치
    - `SettingCoordinator`에 `TrashViewController` 전환하는 메소드 추가
    - 각 뷰컨트롤러마다 네비게이션바를 초기화하는 메서드가 있으므로 이를 extension으로 리팩토링 진행

## 테스트 
- 테스트코드는 따로 없으며, 설정 - 휴지통에서 확인할 수 있습니다.

## 예외 사항 
- 단순히 휴지통화면에 필요한 최소한의 뷰들과 배치만을 구현했습니다.
- 그러므로 삭제 또는 복구하기 위해 셀들을 터치하는 로직은 구현하지 않았습니다. 뿐만 아니라 그후에 발생할 수 있는 상황 또는 로직들을 구현하지 않았습니다. 

### Linked Issue

close #109

gustn3965 committed:
[#109] view: 휴지통화면 셀 구현
- ContentsCollectionViewCell를 리팩토링하여 휴지통화면에 필요한 뷰들을 추가했다

gustn3965 committed:
[#109] view: 휴지통 화면의 상단 헤더뷰 구현

gustn3965 committed:
[#109] view: 휴지통 화면의 하단에 나타나는 삭제 or 복구 버튼 구현
- LeftRightButtonView - 좌측, 우측에 각각 버튼을 가지는 뷰다

gustn3965 committed:
[#109] view: 휴지통 화면 구현
- TrashViewController 를 만들어, 휴지퉁화면에 필요한 뷰들을 배치
- SettingCoordinator에TrashViewController 전환하는 메소드 추가
- 각 뷰컨트롤러마다 네비게이션바를 초기화하는 메서드가 있으므로 이를 extension으로 리팩토링 진행

jwonyLee commented on ThingLog/Extension/UIView+.swift:
````diff
@@ -11,4 +11,40 @@ extension UIView {
+    ///             startPoint: CGPoint(x: 0.0, y: 1.0),
+    ///             endPoint: CGPoint(x: 0.0, y: 0.0)
+    /// ```
+    func setGradient(startColor: UIColor,
````
LGTM :+1: 예시 들어주는 거 좋아요!

jwonyLee commented on ThingLog/Extension/UIView+.swift:
```diff
@@ -11,4 +11,40 @@ extension UIView {
+        gradient.endPoint = endPoint
+        gradient.frame = bounds
+        gradient.name = "customGradient"
+        layer.addSublayer(gradient)
```
😎 이건 그냥 제안입니다! `gradient.name`을 문자열로 직접 넣고 있는데, 파라미터로 `gradientName`을 받아서 설정해주면 좋을 거 같아요~ (where 문은 swiftlint 방식대로 작성해봤어요.)
```swift
func setGradient(startColor: UIColor,
                 endColor: UIColor,
                 startPoint: CGPoint,
                 endPoint: CGPoint,
                 gradientName: String) {
    if let sublayers: [CALayer] = layer.sublayers {
        for sublayer in sublayers where sublayer.name == gradientName {
            return
        }
    }

    let gradient: CAGradientLayer = CAGradientLayer()
    gradient.colors = [startColor.cgColor, endColor.cgColor]
    gradient.locations = [0.0, 1.0]
    gradient.startPoint = startPoint
    gradient.endPoint = endPoint
    gradient.frame = bounds
    gradient.name = gradientName
    layer.addSublayer(gradient)
}
```

jwonyLee commented on ThingLog/ViewController/ContentsCollection/ContentsCollectionViewCell.swift:
````diff
@@ -7,6 +7,18 @@
+/// cell.bottomGradientView.isHidden = false
+/// cell.bottomLabel.isHidden = false
+/// cell.checkButton.isHidden = false
+/// ```
````
:eyes: 휴지통 화면에서 사용하기 위해서 뷰를 외부에 노출시키고 있는데, 메서드를 만들어서 해당 값들을 변경하는 방향으로 하는 건 어떤가요?_?

jwonyLee reviewed (approved):
LGTM :+1:
이 밖에 코멘트로 남기지는 않았지만 문구에서 온점이 빠져있다던가.. 하는 부분이 있는데 이 점은 다시 한 번 확인해주시면 좋을 거 같습니다~

gustn3965 committed:
[#109] 지원님 리뷰 반영
- 온점 최대한 표시
- 휴지통의 화면에서 사용되기 위해 뷰들을 숨기거나 나타내는 Convenience 메소드 추가
- if 대신 where문으로 코드 간략화

gustn3965 commented:
리뷰 감사합니다 ~ 
지원님 피드백 주신 부분들은 최대한 반영했고,
layer에 이름을 받는것은 일단은 보류했습니다. 한 뷰에 그라데이션 레이어는 하나만 들어갈 것 같고, 그라데이션이라고 이미 특정상황으로 지정하기 때문에 이름을 추가적으로 받을 필요성은 아직 못느껴서요!
